packages/jest-reporters/src/GitHubActionsReporter.ts

```diff
@@ -5,55 +5,50 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import {relative} from 'path';
+import slash = require('slash');
 import stripAnsi = require('strip-ansi');
-import type {
-  AggregatedResult,
-  TestContext,
-  TestResult,
-} from '@jest/test-result';
+import type {Test, TestCaseResult} from '@jest/test-result';
+import {
+  getStackTraceLines,
+  getTopFrame,
+  separateMessageFromStack,
+} from 'jest-message-util';
 import BaseReporter from './BaseReporter';
 
-const lineAndColumnInStackTrace = /^.*?:([0-9]+):([0-9]+).*$/;
-
-function replaceEntities(s: string): string {
-  // https://github.com/actions/toolkit/blob/b4639928698a6bfe1c4bdae4b2bfdad1cb75016d/packages/core/src/command.ts#L80-L85
-  const substitutions: Array<[RegExp, string]> = [
-    [/%/g, '%25'],
-    [/\r/g, '%0D'],
-    [/\n/g, '%0A'],
-  ];
-  return substitutions.reduce((acc, sub) => acc.replace(...sub), s);
-}
+const errorTitleSeparator = ' \u203A ';
 
 export default class GitHubActionsReporter extends BaseReporter {
   static readonly filename = __filename;
 
-  override onRunComplete(
-    _testContexts?: Set<TestContext>,
-    aggregatedResults?: AggregatedResult,
+  override onTestCaseResult(
+    test: Test,
+    {failureMessages, ancestorTitles, title}: TestCaseResult,
   ): void {
-    const messages = getMessages(aggregatedResults?.testResults);
-
-    for (const message of messages) {
-      this.log(message);
-    }
+    failureMessages.forEach(failure => {
+      const {message, stack} = separateMessageFromStack(stripAnsi(failure));
+
+      const relativeTestPath = slash(relative('', test.path));
```

Comment:
For some reason, `rootDir` is not pointing to the right place. See https://github.com/facebook/jest/pull/12822#issuecomment-1120434301.

In the other hand, in CI it feels right to keep paths relative to root of the repo.